giles-eco/src/main/java/edu/asu/diging/gilesecosystem/web/core/files/impl/ProcessingRequestsDatabaseClient.java

```diff
@@ -1,77 +1,81 @@
 package edu.asu.diging.gilesecosystem.web.core.files.impl;
 
-import java.util.ArrayList;
 import java.util.List;
+import java.util.stream.Collectors;
 
 import javax.persistence.EntityManager;
-import javax.persistence.PersistenceContext;
-import javax.persistence.TypedQuery;
 
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 import org.springframework.transaction.annotation.Transactional;
 
 import edu.asu.diging.gilesecosystem.septemberutil.properties.MessageType;
 import edu.asu.diging.gilesecosystem.septemberutil.service.ISystemMessageHandler;
-import edu.asu.diging.gilesecosystem.util.exceptions.UnstorableObjectException;
 import edu.asu.diging.gilesecosystem.util.store.objectdb.DatabaseClient;
 import edu.asu.diging.gilesecosystem.web.core.files.IProcessingRequestsDatabaseClient;
 import edu.asu.diging.gilesecosystem.web.core.model.IProcessingRequest;
 import edu.asu.diging.gilesecosystem.web.core.model.impl.ProcessingRequest;
+import edu.asu.diging.gilesecosystem.web.core.repository.ProcessingRequestRepository;
 
 @Transactional
 @Component
 public class ProcessingRequestsDatabaseClient extends DatabaseClient<IProcessingRequest> implements IProcessingRequestsDatabaseClient {
 
-    @PersistenceContext(unitName="entityManagerFactory")
-    private EntityManager em;
+    @Autowired
+    private ProcessingRequestRepository processingRequestRepository;
     
     @Autowired
     private ISystemMessageHandler messageHandler;
 
     @Override
     public List<IProcessingRequest> getRequestByDocumentId(String docId) {
-        TypedQuery<IProcessingRequest> query = getClient().createQuery("SELECT t FROM " + IProcessingRequest.class.getName()  + " t WHERE t.documentId = '" + docId + "'", IProcessingRequest.class);
-        return query.getResultList();
+        List<ProcessingRequest> processingRequests = processingRequestRepository.findByDocumentId(docId);
+        return processingRequests.stream().map(processingRequest -> (IProcessingRequest) processingRequest).collect(Collectors.toList());
     }
     
     @Override
     public List<IProcessingRequest> getProcRequestsByRequestId(String procReqId) {
-        return searchByProperty("requestId", procReqId, ProcessingRequest.class);
+        List<ProcessingRequest> processingRequests = processingRequestRepository.findByRequestId(procReqId);
+        return processingRequests.stream().map(processingRequest -> (IProcessingRequest) processingRequest).collect(Collectors.toList());
     }
     
     @Override
     public void saveNewRequest(IProcessingRequest request) {
         request.setId(generateId());
         try {
-            store(request);
-        } catch (UnstorableObjectException e) {
-            // should never happen
+            processingRequestRepository.save((ProcessingRequest) request);
+        } catch (IllegalArgumentException e) {
```

Comment:
in other classes this exception is thrown instead of caught; this should be consistent.